ironfish/src/networkDefinition.ts

```diff
@@ -47,3 +51,61 @@ export const networkDefinitionSchema: yup.ObjectSchema<NetworkDefinition> = yup
       .defined(),
   })
   .defined()
+
+export async function getNetworkDefinition(
+  config: Config,
+  internal: InternalStore,
+  files: FileSystem,
+): Promise<NetworkDefinition> {
+  let networkDefinitionJSON = ''
+
+  // Try fetching custom network definition first, if it exists
+  if (config.isSet('customNetwork')) {
+    networkDefinitionJSON = await files.readFile(files.resolve(config.get('customNetwork')))
+  } else {
+    let networkId = internal.get('networkId')
+
+    if (
+      internal.isSet('networkId') &&
+      config.isSet('networkId') &&
+      networkId !== config.get('networkId')
+    ) {
+      throw Error('Network ID flag does not match network ID stored in datadir')
+    }
+
+    if (config.isSet('networkId')) {
+      networkId = config.get('networkId')
+    }
```

Comment:
This is kind of just a code style thing, but I'd suggest defining networkId after the `if` check, because I think it makes it easier to read if networkId can be declared as `const`. Like this:

```suggestion
    if (
      internal.isSet('networkId') &&
      config.isSet('networkId') &&
      internal.get('networkId') !== config.get('networkId')
    ) {
      throw Error('Network ID flag does not match network ID stored in datadir')
    }
    
    const networkId = config.isSet('networkId') ? config.get('networkId') : internal.get('networkId')
```